source/agora/consensus/Ledger.d

```diff
@@ -936,42 +936,42 @@ public class Ledger
 
     ***************************************************************************/
 
-    private string validateBlockSignature (in Block block) @safe nothrow
+    public string validateBlockSignature (in BlockHeader header) @safe nothrow
```

Comment:
This whole diff is a refactoring (a good one though), as it reduces the dependency from `Block` to `BlockHeader`. Could you split it in its own commit ?